provider/interface.go

```diff
@@ -6,12 +6,38 @@ import (
 	"github.com/pivotal-cf/brokerapi/domain"
 )
 
-//go:generate counterfeiter -o fakes/fake_service_provider.go . ServiceProvider
-type ServiceProvider interface {
-	Provision(context.Context, ProvisionData) (dashboardURL, operationData string, isAsync bool, err error)
-	Deprovision(context.Context, DeprovisionData) (operationData string, isAsync bool, err error)
-	Bind(context.Context, BindData) (binding domain.Binding, err error)
-	Unbind(context.Context, UnbindData) (unbinding domain.UnbindSpec, err error)
-	Update(context.Context, UpdateData) (operationData string, isAsync bool, err error)
-	LastOperation(context.Context, LastOperationData) (state domain.LastOperationState, description string, err error)
+type Provider interface{}
```

Comment:
are literally all operations optional? (I mean, I guess the answer is "yes" in the case of the observe service broker, but it's not exactly a great example.)

I would have thought that Provision/Deprovision would be minimum required operations?

Reply:
Yeah I'm on the fense about that one ... I think `Provider` should probably contain `Provisioner` ... I don't like the empty interface either

Reply:
see other comment on combining interfaces